civihr_employee_portal/src/Service/EmergencyContactDeletionService.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Drupal\civihr_employee_portal\Service;
+
+use CRM_Hremergency_Service_EmergencyContactService as EmergencyContactService;
+
+class EmergencyContactDeletionService {
+
+  /**
+   * Deletes an emergency contact and sends a notification mail about it
+   *
+   * @param int $id
+   *   The ID of the emergency contact
+   */
+  public function delete($id) {
+    $service = new EmergencyContactService();
+    $emergencyContact = $service->find($id);
+
+    if (!$emergencyContact) {
+      $err = sprintf('Emergency contact with ID "%d" was not found', $id);
+      throw new \Exception($err);
+    }
+
+    if (!$this->canDelete($emergencyContact)) {
+      $err = 'You do not have permission to delete that emergency contact';
+      throw new \Exception($err);
+    }
+
+    $service->delete($id);
```

Comment:
Is there any reason for using the `EmergencyContactService` directly instead of calling the `Contact.deleteEmergencyContact` API? I think that using the API would be better here for two reasons:
- We avoid exposing implementation details of the `hremergency` extension to the Drupal module (I know we already do this in other places, but we should avoid it when possible)
- This class is duplicating all the logic we already have [here](https://github.com/compucorp/civihr/blob/staging/org.civicrm.hremergency/api/v3/Contact/Deleteemergencycontact.php#L25-L43)

Reply:
Good point, I forgot it was me that added that endpoint! I'd still like to use the `EmergencyContactService` to fetch the emergency contact details though, because it helps a lot in getting back a entity with logical property names (using the custom field name) instead of `name_12` etc.